daemon/api_apps.go

```diff
@@ -223,10 +225,78 @@ func getLogs(c *Command, r *http.Request, user *auth.UserState) Response {
 
 var servicestateControl = servicestate.Control
 
-func postApps(c *Command, r *http.Request, user *auth.UserState) Response {
+func decodeServiceInstruction(body io.ReadCloser, u *user.User) (*servicestate.Instruction, error) {
 	var inst servicestate.Instruction
-	decoder := json.NewDecoder(r.Body)
+	decoder := json.NewDecoder(body)
 	if err := decoder.Decode(&inst); err != nil {
+		return nil, err
+	}
+	return &inst, nil
+}
+
+func hasUserService(apps []*snap.AppInfo) bool {
+	for _, app := range apps {
+		if app.IsService() && app.DaemonScope == snap.UserDaemon {
+			return true
+		}
+	}
+	return false
+}
+
+var systemUserFromRequest = func(r *http.Request) (*user.User, error) {
+	uid, err := uidFromRequest(r)
+	if err != nil {
+		return nil, err
+	}
+	return user.LookupId(strconv.Itoa(int(uid)))
```

Comment:
I'm a bit paranoid  but we should check that result has Uid == "0" only if the input had uid == 0